test-grid-scripts/bbg/kafka/deployment-kafka-service.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+# Copyright (c) 2019, WSO2 Inc. (http://wso2.org) All Rights Reserved.
+#
+# WSO2 Inc. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+readonly deployment_kafka_parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+readonly deployment_kafka_grand_parent_path=$(dirname ${deployment_kafka_parent_path})
+readonly deployment_kafka_great_grand_parent_path=$(dirname ${deployment_kafka_grand_parent_path})
+
+. ${deployment_kafka_great_grand_parent_path}/common/usage.sh
+. ${deployment_kafka_great_grand_parent_path}/setup/setup_deployment_env.sh
+
+
+function clone_bbg_and_set_bal_path() {
+    local bbg_repo_name="messaging-with-kafka"
+    git clone https://github.com/ballerina-guides/messaging-with-kafka.git --branch kafka-testgrid
+    bal_path_admin=${bbg_repo_name}/guide/product_admin_portal/product_admin_portal.bal
+}
+
+function setup_deployment() {
+    clone_bbg_and_set_bal_path
+    download_and_install_kafka_connector
+    deploy_kafka_resources
+    replace_variables_in_bal_file $bal_path_admin
+    build_and_deploy_guide
+    kubectl describe svc
+    wait_for_pod_readiness
+    retrieve_and_write_properties_to_data_bucket
+    local is_debug_enabled=${infra_config["isDebugEnabled"]}
+    if [ "${is_debug_enabled}" = "true" ]; then
+        print_kubernetes_debug_info
+    fi
+}
+
+## Functions
+function download_and_install_kafka_connector() {
+    KAFKA_CONNECTOR_VERSION=0.980.2
```

Comment:
Can we pass this through testgrid.yaml instead of hard-coding?

Reply:
Done